beacon-chain/blockchain/service.go

```diff
@@ -299,6 +405,46 @@ func (c *ChainService) processBlock(block *types.Block) error {
 	return nil
 }
 
+func (c *ChainService) processBlockNew(block *types.Block) error {
+
+	blockhash, err := block.Hash()
+	if err != nil {
+		return fmt.Errorf("could not hash incoming block: %v", err)
+	}
+
+	beaconState, err := c.beaconDB.GetState()
+	if err != nil {
+		return fmt.Errorf("failed to get beacon state: %v", err)
+	}
+
+	if block.SlotNumber() == 0 {
```

Comment:
Is this correct? a genesis block still has to go through `processBlockNew` though? We just can't receive a genesis block

Reply:
Actually a genesis block is not processed, each client will come with a common genesis block and genesis state.